angularjs-viewhead.js

```diff
@@ -1,76 +1,66 @@
-
 (function (angular) {
+    'use strict';
+
+    var mod = angular.module('viewhead', []);
+
+    // Data service that is shared between all directives
+    mod.factory('viewTitleDataService', function () {
+        return {};
+    });
+
+    mod.directive('viewTitle', ['$rootScope', 'viewTitleDataService', function ($rootScope, viewTitleDataService) {
+        return {
```

Comment:
Rather than having the separate `viewTitleDataService` service, could we instead just have a local variable here inside this factory function, which would then be part of the closure for the `link` function and shared between all instances? I'm actually a bit rusty on Angular since I've not used it in six months or so, but IIRC this factory function is called once per injector, meaning that any local variables it creates should behave effectively as singletons within a given injector.

This is a relatively minor thing, but this way we'd prevent other directives and services from accessing this private data structure, I think is worth doing if it's easy.